<launch>

	<arg name="quad"/>
	<arg name="bag" default="false"/>
	<arg name="memory" default="true"/>

	<node ns="$(arg quad)" name="tip" pkg="tip" type="tip" output="screen" clear_params="true">
        <param name="~use_memory" value="$(arg memory)" type="bool"/>

		<rosparam file="$(find tip)/param/tip.yaml" command="load"/>

		<remap from="~state" to="state"/>
		<remap from="~event" to="event" />
		<remap from="~global_goal" to="global_goal" />
		<remap from="~camera/cloud" to="camera/cloud"/>	
		<remap from="~mode" to="mode"/>	

		<remap from="~traj" to="traj"/>
		<remap from="~tip_data" to="tip_data"/>
		<remap from="~goal" to="goal" />
		<remap from="~clouds" to="clouds"/>	
		<remap from="~boundary_layer" to="boundary_layer"/>	
	</node>

	<group if="$(arg bag)">
        <!-- Ensure RosBag is saving important data -->
        <node ns="$(arg quad)" pkg="rosbag" type="record" name="rosbag_record_planner" args="-o /media/ubuntu/LOGGER_USB/bags/$(arg quad)PlannerLogFile global_goal new_global_goal traj event tip_data /tf"/>
    </group>

</launch>
